docs/cni-plugin.md

```diff
@@ -35,16 +35,29 @@ Another example with a trunk port and jumbo frames:
 }
 ```
 
+Another example with QinQ configured:
+
+```json
+{
+    "name": "myqinqnet",
+    "type": "ovs",
+    "bridge": "mynet2",
+    "vlan": 1000,
+    "cvlan": [ { "id" : 24 }, { "minID" : 100, "maxID" : 102 } ]
```

Comment:
I agree with the former. For the "two types" part, I'd prefer to keep it this way since it is already a format used for the trunk, let's reuse it (with all the helper functions). We can consider changes in the overall API in a follow-up.

Reply:
@EdDev @phoracek Thank you for pointing this out. Do you suggest adding additional parameter that would indicate whether to enable QinQ? Or do you think it's enough to use `cvlan` for this and allow passing empty list to include the `allow all to pass` use case?

Reply:
Good question.

We can introduce `vlanMode` attribute to mimic ovs-vsctl. Let's stick to the OVS database schema:

```
       VLAN Configuration:
         vlan_mode                   optional  string,  one  of  access, dot1q-tunnel, native-tagged,
                                     native-untagged, or trunk
         tag                         optional integer, in range 0 to 4,095
         trunks                      set of up to 4,096 integers, in range 0 to 4,095
         cvlans                      set of up to 4,096 integers, in range 0 to 4,095
         other_config : qinq-ethtype
                                     optional string, either 802.1ad or 802.1q
         other_config : priority-tags
                                     optional string, one of always, if-nonzero, or never
```

Details:

```
       vlan_mode:  optional  string,  one of access, dot1q-tunnel, native-tagged, native-untagged, or       trunk
              The VLAN mode of the port, as described above. When this column  is  empty,  a  default
              mode is selected as follows:

              ·      If tag contains a value, the port is an access port. The trunks column should be
                     empty.

              ·      Otherwise, the port is a trunk port. The trunks column value is honored if it is
                     present.
```

```
       cvlans: set of up to 4,096 integers, in range 0 to 4,095
              For  a dot1q-tunnel port, the customer VLANs that this port includes. If this is empty,
              the port includes all customer VLANs.

              For other kinds of ports, this setting is ignored.
```

(see man `ovs-vswitchd.conf.db`)

Our API would need some rework to fully fit this. Let's keep it simple while doing QinQ. We should require `vlanMode` to be set to `dot1q-tunnel`. If it is, we default to all cvlans or use defined range. If it is not set (or set to something else) and cvlans is set, we should return an error message.

Reply:
Updated

Reply:
> I agree with the former. For the "two types" part, I'd prefer to keep it this way since it is already a format used for the trunk, let's reuse it (with all the helper functions). We can consider changes in the overall API in a follow-up.

I understood there is no rush with this QnQ option, therefore, I think it will be preferred to do it right from start and not change the API later on.
We can brainstorm it briefly on an API change issue. It should not take too long.

Reply:
Yeah, let's meet and discuss this.